src/ServiceControl/Infrastructure/Extensions/QueryableExtensions.cs

```diff
@@ -161,11 +169,52 @@ public static IDocumentQuery<FailedMessageViewIndex.SortAndFilterOptions> Filter
             }
             sb.Append(")");
 
+            source.AndAlso();
```

Comment:
This is a terrible API. What does this do? Should we add a comment here to explain why we need this?